### ISSUE TITLE
BooleanExpressionComplexity misidentifies integer expression as boolean expression

### ISSUE THREAD

**Issue by jasperreijer** [checkstyle/checkstyle#572]
this expression (names changed to protect the guilty):
new Settings( Settings.RESIZABLE | Settings.SCROLLBARS | Settings.LOCATION_BAR | Settings.MENU_BAR | Settings.TOOL_BAR);

where  

public Settings(int flags);
 
Causes a major violation of BooleanExpressionComplexity. However, this expression is not Boolean or boolean (http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.22).

**Comment by romani** [comment]
BooleanExpressionComplexity, should highlight only boolean expressions, arguments of method calls should be ignored as checkstyle does not know types of methods from different classes. That should be described in javadoc.

**Comment by terje2001** [comment]
We believe the bug in BooleanExpressionComplexity above also manifests itself under other conditions too, like multi catches:
```
try {
  ....
} catch (Exception | Error e) {
  ....
}
```
This causes an NPE as described above as well. Ugly work-around for us now is to add the property token like this to the config:
```
<module name="BooleanExpressionComplexity">
  <property name="tokens" value="BXOR,LAND,LOR"/>
</module>
```
Please fix soon! :-)

Thanks!

**Comment by ychulovskyy** [comment]
NPE has been fixed by #654

**Comment by romani** [comment]
NPE is fixed but original issue is still there.

**Comment by romani** [comment]
resolved will be in 6.4
